core/services/relay/evm/evm.go

```diff
@@ -373,7 +371,7 @@ func newConfigProvider(lggr logger.Logger, chain legacyevm.Chain, opts *types.Re
 	return newConfigWatcher(lggr, aggregatorAddress, contractABI, offchainConfigDigester, cp, chain, relayConfig.FromBlock, opts.New), nil
 }
 
-func newContractTransmitter(lggr logger.Logger, rargs commontypes.RelayArgs, transmitterID string, configWatcher *configWatcher, ethKeystore keystore.Eth) (*contractTransmitter, error) {
+func newContractTransmitter(lggr logger.Logger, rargs commontypes.RelayArgs, transmitterID string, configWatcher *configWatcher, ethKeystore keystore.Eth, gasLimit uint32) (*contractTransmitter, error) {
```

Comment:
i'd rather this this implemented with a new opts struct than a explicit argument.

type configTransmitterOpts struct {
configWatcher *configWatcher
// override the gas limit default provided in the config watcher
gasLimit uint32
}

func newContractTransmitter (..., opts configTransmitterOpts) {
...
gasLimit := configWatcher.chain.Config().EVM().GasEstimator().LimitDefault()
if opts.gasLimit != 0 {
	gasLimit = opt.gasLimit
}

}

all the call site need to be updated, but only the automation call site needs to explicitly set a gasLimit option.

this implementation seems more inline with the intention of passing a *configwatcher and deriving relevant options/configuration from it